### ISSUE TITLE
verifyInOrder not working and no example test

### ISSUE THREAD

**Issue by kuhnroyal** [felangel/mocktail#25]
**Describe the bug**
I tried migrating some simple mockito tests and got pretty much everything working except my `verifyInOrder([])` calls.

I am not sure what to pass as arguments there. I tried passing the calls directly and also as closure but neither of these approaches worked.

**Comment by felangel** [comment]
Hi @kuhnroyal 👋 
Thanks for opening an issue!

This should be fixed in [v0.0.2-dev.3](https://pub.dev/packages/mocktail) 👍 

Example usage:

```dart
when(() => cat.sound()).thenReturn('meow');
when(() => cat.name).thenReturn('fluffy');
cat..sound()..name;
verifyInOrder(() => [cat.sound(), cat.name]);
```

Hope that helps and sorry for the inconvenience!

**Comment by kuhnroyal** [comment]
Thanks for your work!

Not sure that fixes it, try adding `verifyNoMoreInteractions(cat)` at the end.
It seems only the first parameter is evaluated or counted.

**Comment by felangel** [comment]
Can you share a reproduction sample?

**Comment by kuhnroyal** [comment]
I am trying this but I don't have a full example that I can share.

```dart
      final migration = MockMigration();

      when(() => migration.migrate2(database, migrator)).thenAnswer((_) => Future<void>.value());
      when(() => migration.migrate3(database, migrator)).thenAnswer((_) => Future<void>.value());
      when(() => migration.migrate4(database, migrator)).thenAnswer((_) => Future<void>.value());
      when(() => migration.migrate5(database, migrator)).thenAnswer((_) => Future<void>.value());

      await Migrations(
        migrations: <int, MigrationFunction<Database>>{
          2: (Database db, Migrator m) => migration.migrate2(db, m),
          3: (Database db, Migrator m) => migration.migrate3(db, m),
          4: (Database db, Migrator m) => migration.migrate4(db, m),
          5: (Database db, Migrator m) => migration.migrate5(db, m),
        },
      ).onUpgrade(database, migrator, 1, 5);

      // verify(() => migration.migrate2(database, migrator)).called(1);
      // verify(() => migration.migrate3(database, migrator)).called(1);
      // verify(() => migration.migrate4(database, migrator)).called(1);
      // verify(() => migration.migrate5(database, migrator)).called(1);

      verifyInOrder(() => [
            migration.migrate2(database, migrator),
            migration.migrate3(database, migrator),
            migration.migrate4(database, migrator),
            migration.migrate5(database, migrator),
          ]);

      verifyNoMoreInteractions(migration);
```

**Comment by felangel** [comment]
@kuhnroyal sorry for the inconvenience! I realized the issue is each verification needs to be wrapped in it's own closure in order to evaluate them all. Just included the fix in v0.0.2-dev.4 and the updated usage should look like:

```dart
final migration = MockMigration();

when(() => migration.migrate2(database, migrator)).thenAnswer((_) => Future<void>.value());
when(() => migration.migrate3(database, migrator)).thenAnswer((_) => Future<void>.value());
when(() => migration.migrate4(database, migrator)).thenAnswer((_) => Future<void>.value());
when(() => migration.migrate5(database, migrator)).thenAnswer((_) => Future<void>.value());

await Migrations(
migrations: <int, MigrationFunction<Database>>{
    2: (Database db, Migrator m) => migration.migrate2(db, m),
    3: (Database db, Migrator m) => migration.migrate3(db, m),
    4: (Database db, Migrator m) => migration.migrate4(db, m),
    5: (Database db, Migrator m) => migration.migrate5(db, m),
},
).onUpgrade(database, migrator, 1, 5);

// verify(() => migration.migrate2(database, migrator)).called(1);
// verify(() => migration.migrate3(database, migrator)).called(1);
// verify(() => migration.migrate4(database, migrator)).called(1);
// verify(() => migration.migrate5(database, migrator)).called(1);

verifyInOrder([
    () => migration.migrate2(database, migrator),
    () => migration.migrate3(database, migrator),
    () => migration.migrate4(database, migrator),
    () => migration.migrate5(database, migrator),
]);

verifyNoMoreInteractions(migration);
```

Hope that helps!

**Comment by kuhnroyal** [comment]
Thanks a lot! Everything is working now.
